src/components/primitives/Forms/AddressInput.tsx

```diff
@@ -17,35 +18,34 @@ export const AddressInput: React.FC<InputProps<string>> = ({
   onChange,
   isInvalid,
   disabled = false,
+  defaultValue,
   ...rest
 }) => {
   const { imageUrl } = useENSAvatar(value, MAINNET_ID);
   const shouldShowAvatar = !!isAddress(value) || value?.endsWith('.eth');
+  const [disabledState, setDisabled] = useState(defaultValue ? true : disabled);

```

Comment:
We should use the useState convention, like:

`[disabledState, setDisabledState]`